Backwardcompatibility

jlamaille opened the pull request:
The goal is to add the notion of non-retro compatibility of changes with a visible rendering on the html part.

I added unit tests.

I still have the readme to update I do it right now.

jlamaille commented:
You can also see the non-backwards compatible changes

jlamaille commented:
Okay to merge for me.

Sayi commented:
Hi, You are awesome!

I have seen the code submitted many times and you modified 40+ files, can you describe in detail which features have been added?

jlamaille commented:
Hi,

Thank you.
- I added the notion of backwards compatibility check or not of the interface contract.
- Added lombok for the auto generated getter and setter.
- A few small renames 
- Add unit tests

I think that's all.

On the other hand I see that the maven install is done with an open jdk 11, it's normal? because it looks like it's stuck on lombok precisely.

jlamaille commented:
I'm thinking of adding an option on the standard output to bring up the notion of backward compatibility or not if you want to go through the command.
I can make a pull request if this one is merged in the meantime.

jlamaille commented:
finally I think the use of the json format is enough.
I think a proofreading of the code would be welcome before the mege.
I'm available to fix if things go wrong.

jlamaille commented:
<img width="290" alt="image" src="https://user-images.githubusercontent.com/13170055/86447888-4606c380-bd16-11ea-8a5e-31eb0b8fa09c.png">

The build does not pass with the lombok 1.16.X version compatible with jdk8
It passes with the lombok 1.18.X because it builds in travis with an openJdk 11.

On the other hand the project is declared in jdk1.8 cf my capture so I don't know what to do :D
